### PR TITLE
shouldOpenCalendar prop for Agenda

### PR DESCRIPTION
**Pull request by shuzootani**
### added `shouldOpenCalendar` prop for `<Agenda />`
this allows you to control a flag to show CalendarView by passing a prop to `<Agenda />`

it makes more sense to toggle CalendarView depending on `shouldOpenCalendar` prop rather than calling `chooseDay()`.

related to #333 #277 #227 

## use cases
- for the simple case like #277 
if you wanna show calendar view only at initial render,
you can just pass `shouldOpenCalendar: true` to `<Agenda />`

- for the case like #333 
suppose you're using [react-native-navigation](https://github.com/wix/react-native-navigation) 
and you wanna show CalendarView not just when the component is mounted, but also whenever the user comes back to the screen where `<Agenda />` is rendered.

``` js
import { Navigation } from 'react-native-navigation'

constructor() {
    this.state = { shouldOpenCalendar: false }
    Navigation.events().bindComponent(this);
}

componentDidAppear() {
    this.setState({ shouldOpenCalendar: true })
}

<Agenda shouldOpenCalendar={this.state.shouldOpenCalendar} />

```

### CODE REVIEW

File: src/agenda/index.js [hunk 1]
```diff
@@ -135,7 +137,9 @@ export default class AgendaView extends Component {
     // When user touches knob, the actual component that receives touch events is a ScrollView.
     // It needs to be scrolled to the bottom, so that when user moves finger downwards,
     // scroll position actually changes (it would stay at 0, when scrolled to the top).
-    this.setScrollPadPosition(this.initialScrollPadPosition(), false);
+    if (!this.props.shouldOpenCalendar) {
```
Comment (shuzootani) [inline src/agenda/index.js]: prevent an animation conflict on initial render so that it either sets initial position or open the calendar

**Comment by shuzootani** [comment]
for naming, which do you prefer `isCalendarOpen` or `shouldOpenCalendar`?

**Comment by raen79** [comment]
👍

**Comment by shuzootani** [comment]
@raen79 Did you test my branch?
This is just my PR for this feature, which is not merged yet!

**Comment by Inbal-Tish** [comment]
@shuzootani Hi. Thank you for your contribution! 
I like the idea behind this PR, but I would like to suggest a modification to 'initialPosition' (see ExapandableCalendar) instead of 'shouldOpenCalendar' to control the calendar's position (open/closed). First, since it is clearer to the user and, second, for keeping the API consistent. 
Hope you can make the changes. Thanks!

**Pull request closed without merging**